contracts/0.4.24/Lido.sol

```diff
@@ -999,10 +936,168 @@ contract Lido is ILido, StETH, AragonApp {
     }
 
     /**
-    * @dev Size-efficient analog of the `auth(_role)` modifier
-    * @param _role Permission name
-    */
+     * @dev Set max positive token rebase value
+     * @param _maxPositiveTokenRebase max positive token rebase, nominated in MAX_POSITIVE_REBASE_PRECISION_POINTS
+     */
+    function _setMaxPositiveTokenRebase(uint256 _maxPositiveTokenRebase) internal {
+        MAX_POSITIVE_TOKEN_REBASE_POSITION.setStorageUint256(_maxPositiveTokenRebase);
+
+        emit MaxPositiveTokenRebaseSet(_maxPositiveTokenRebase);
+    }
+
+    /**
+     * @dev Size-efficient analog of the `auth(_role)` modifier
+     * @param _role Permission name
+     */
     function _auth(bytes32 _role) internal view auth(_role) {
         // no-op
     }
+
+    /**
+     * @dev Invokes a deposit call to the Staking Router contract and updates buffered counters
+     * @param _maxDepositsCount max deposits count
+     * @param _stakingModuleId id of the staking module to be deposited
+     * @param _depositCalldata module calldata
+     */
+    function deposit(uint256 _maxDepositsCount, uint256 _stakingModuleId, bytes _depositCalldata) external {
+        ILidoLocator locator = getLidoLocator();
+
+        require(msg.sender == locator.depositSecurityModule(), "APP_AUTH_DSM_FAILED");
+        require(_stakingModuleId <= uint24(-1), "STAKING_MODULE_ID_TOO_LARGE");
+        _whenNotStopped();
+
+        IWithdrawalQueue withdrawalQueue = IWithdrawalQueue(locator.withdrawalQueue());
+        require(!withdrawalQueue.isBunkerModeActive(), "CANT_DEPOSIT_IN_BUNKER_MODE");
```

Comment:
need to include this constraint into `DepositSecurityModule.canDeposit`

Reply:
```
Lido.canDeposit = ()=> (!isBunkerModeActive && _whenNotStopped)

Reply:
done